filesystem/Upload.php

```diff
@@ -208,6 +208,7 @@ public function load($tmpFile, $folderPath = false) {
 			$this->file->ParentID = $parentFolder ? $parentFolder->ID : 0;
 			// This is to prevent it from trying to rename the file
 			$this->file->Name = basename($relativeFilePath);
+			$this->file->Title = preg_replace("/{$fileSuffix}$/", '', $tmpFile['name']);
```

Comment:
@dhensby 
Potentially filename can be file.jpgname.jpg
str_replace is not suitable in this case:
regular expression is needed for check extension only in the end of string.

I have no idea how to correctly escape $fileSuffix for regexp.
Can you advice some solution?